test/mix_java_scala_lib.bzl

```diff
@@ -0,0 +1,11 @@
+def _mix_java_scala_lib_impl(ctx):
+    paths = [f.path for f in ctx.files.deps]
+    for path in paths:
+        print(path)
```

Comment:
I think having an output file with this content is cleaner than using print but it also does the work so leaving it up to you (we’ve had quite a few iterations on this change unfortunately)

Reply:
If I have an output file from here, how do I test that it contains the right jars? And if I do that, why not test the `scala_library` `MixJavaScalaLib`'s outputs directly?

Reply:
Two ways:
1. Have a scala_test target that depends on this output as data and asserts its contents (pro- no bash, incremental bazel run)
2. Have your current e2e assert against the content of the file (pro- feels less brittle than asserting on debug prints).
In any case I'm ok with the current implementation. I think it can be better but I don't feel strongly about it. Your call.

Reply:
@ittaiz Now I understand what you're saying. Option 1 seems like a good path forward, I'll make the change. Thanks!